### ISSUE TITLE
Multiple window IDs

### ISSUE THREAD

**Issue by jdeyton** [visit-vis/visit_java_client#15]
=== Brief Summary ===
Trying to open separate windows with unique IDs using the same `VisItSwtConnection` does not appear to work properly.

=== Note ===
The easiest place to debug this on the ICE side is to open up the branch jordan/VisItVizService (this will soon be in the master branch). In the bundle org.eclipse.ice.viz.service.visit, there is a class `VisItConnectionAdapter` with a method `public int getNextWindowId()`. At the moment, the code in that method just returns 1 for the window ID, but the commented out code starts at 0 and looks for the lowest-indexed available window ID according to the `VisItSwtConnection`. Just uncomment the code there. You can also set a breakpoint in that method if you like.

=== Reproducing ===
To see what happens when the code tries to use different window IDs, do the following:

0. Throw a mesh file that VisIt can load into ICEFiles/default. I have only used Exodus files so far.
1. Launch ICE.
2. Open Window -> Preferences and go to the preference node Visualization -> VisIt. Add a new local connection for VisIt and set its path (the defaults everywhere else are fine, and the host OS is ignored). Hit OK. These preferences are then saved for future runs.
3. In the Item View, create a new "MOOSE Model Builder".
4. Select the MOOSE app "BISON" and click save.
5. In the MOOSE tree (it may be called something like Input Data Tree, and may be at the bottom of the window), select the node "Mesh".
6. In the Properties View (probably behind the MOOSE tree now), change the type to "FileMesh". Then update the "File" value to the name of the mesh file you put into ICEFiles/default, including its extension.
7. Press enter and click save.
8. In the MOOSE Model Builder view, change it to the "Mesh" tab (the tabs are at the bottom in Linux and Windows, not sure about OSX).
9. Open the Resources View and double-click the mesh file.
10. VisIt should render the mesh inside the MOOSE Model Builder.

=== Results ===
Now, if you *did not* uncomment the code in the `VisItConnectionAdapter`, then what happens is when you try double-clicking the mesh file again, the same VisIt window is displayed again in the MOOSE Model Builder. This is fine, except anything done to one of the views also changes the other (mouse dragging, zooming, setting the plot type through the right-click menu, etc.)

If you *did* uncomment the code in the `VisItConnectionAdapter`, then the `VisItSwtWidget` just doesn't show anything the first or second time you opened the mesh file. You can set a breakpoint in that method to verify that the window IDs are different.

For reference, the code that creates the `VisItSwtWidget` is actually in the same bundle in `VisItPlotRender#createPlotComposite(...)`.

I'm not sure if I'll have much time to look at this problem specifically, but I'll try to help where I can.

**Comment by HarinarayanKrishnan** [comment]
Hi Jordan,

Thanks for letting me know. I'll investigate the issue.

Best regards,
Hari

**Comment by jdeyton** [comment]
You shouldn't need to check out a separate branch for this any more. All of the code mentioned above is part of ICE's master branch.

**Comment by jdeyton** [comment]
When I print out the JSON objects in the VisItThread (in VisItProxy), here's a small portion of what I get:

~~~~
VisIt message: JSON received: {"contents":[168,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,false],"id":47,"metadata":[4,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,1],"typename":"ViewerRPC"}
VisIt message: JSON received: {"contents":[202,1,0,0,"","C:\\Users\\djg\\ICEFiles\\default\\coarse10_rz.e","","",[],0,0,[0,0],0,0,0,"",[],[],[],"","",[0,0,0],0,0,true,0,0,0,"{action:RegisterNewWindow,windowId:2}","",1,{"data":null,"metadata":"0"}],"id":47,"metadata":[4,4,4,4,8,8,8,8,16,4,4,12,4,4,4,8,12,12,12,8,8,15,4,4,1,4,4,4,8,8,4,17],"typename":"ViewerRPC"}
VisIt message: JSON received: {"contents":[168,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,true],"id":47,"metadata":[4,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,1],"typename":"ViewerRPC"}
VisIt message: JSON received: {"contents":[232833286],"id":2,"metadata":[4],"typename":"SyncAttributes"}
VisIt message: JSON received: {"contents":[168,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,false],"id":47,"metadata":[4,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,1],"typename":"ViewerRPC"}
VisIt message: JSON received: {"contents":[165,1,2,0,"","C:\\Users\\djg\\ICEFiles\\default\\coarse10_rz.e","","",[],0,0,[0,0],0,0,0,"",[],[],[],"","",[0,0,0],0,0,true,1340,1020,0,"{action:RegisterNewWindow,windowId:2}","",1,{"data":null,"metadata":"0"}],"id":47,"metadata":[4,4,4,4,8,8,8,8,16,4,4,12,4,4,4,8,12,12,12,8,8,15,4,4,1,4,4,4,8,8,4,17],"typename":"ViewerRPC"}
VisIt message: JSON received: {"contents":["The specified window does not exist.",[],false,0],"id":3,"metadata":[8,11,1,4],"typename":"MessageAttributes"}
~~~~

It looks like the RegisterNewWindow call is being sent successfully, but for whatever reason we're getting back a (lot of) message(s) that "The specified window does not exist."

**Comment by jdeyton** [comment]
Are these calls ultimately going through the VisIt python API after being se

If you think the manuals here (https://wci.llnl.gov/simulation/computer-codes/visit/manuals) are up-to-date enough for me to take a stab at this, let me know. I'd like to get this fixed if possible this week or next.

**Comment by HarinarayanKrishnan** [comment]
Not quite. I am having a few machine related difficulties, my current one
is getting fixed and a new one hasn't arrived yet. I am trying to get ICE
rebuilt on my laptop at the moment. Once that is done, I can look into this
issue.

The way the calls should normally work is you create a window the register
it.

Something like:
AddWindow() <-- adds a window on the server.
RegisterNewWindow() <-- registers it to be sent to the remote client
